pkg/transport/manager.go

```diff
@@ -216,18 +217,29 @@ func (tm *Manager) SaveTransport(ctx context.Context, remote cipher.PubKey, tpTy
 
 	var err error
 	for i := 0; i < tries; i++ {
-
 		mTp, err := tm.saveTransport(remote, tpType)
 		if err != nil {
 			return nil, err
 		}
 
 		if err = mTp.Dial(ctx); err != nil {
+			// TODO(nkryuchkov): Check for an error that underlying connection is not established
+			// and try again in this case. Otherwise, return the error.
+			pkTableErr := fmt.Sprintf("pk table: entry of %s does not exist", remote.String())
```

Comment:
In my opinion, we should make our errors implement `net.Error` and make use of the `.Permanent() bool` method here. If `.Permanent() == true`, we stop serving.

Reply:
@evanlinjin Good idea, but I think it might be quite out of the scope of this PR. What do you think of leaving it as is for now and modifying after your proposal is implemented?